Safrul Lauparenta

safruls opened the pull request:
- Create an API documentation as of now.
- Later tonight,  I will add and commit the corresponding REST API (includes register and login using JWT)

.env:
SECRET='secret123'
CLIENT_ID=411524848726-8596p4ossfobk8fd2c3sv0acalu7mpvj.apps.googleusercontent.com

safruls committed:
API Documentation for A Fancy Todo App

safruls committed:
Edit API documentation

safruls committed:
add REST API

safruls committed:
Add access token, authentication and authorization middleware. Use dotenv for JWT token methods

safruls committed:
Add errorHandler.js and move the authentication and authorization function into a file in middlewares

safruls committed:
Add deezer as 3rd Party API. Update description with some music related to the title using query result

safruls committed:
On client:
1. Add navbar, login form, and register form on index,html.
2. Add functions corresponding to navbar.
3. Add functions to enable user to login and register from client side.
On server:
1. Minor revision on apiController.js

safruls committed:
On client side:
1. Add create todo form, edit form (whole fields), edit form (status field) on index.html
2. Add corressponding functions to allow CRUD capabilities from client side on main.js
3. Add error message on index.html to display any error messages.

ziadyjo commented on server/api-doc.md:
```diff
@@ -0,0 +1,272 @@
+- GET /todos/:id
+- PUT /todos/:id
+- PATCH /todos/:id
+- DELETE /todos/:id
```
di update lagi

safruls replied:
Ok kak!

ziadyjo commented on server/middlewares/errorHandler.js:
```diff
@@ -0,0 +1,41 @@
+const errorHandler = (err, req, res, next) => {
```
udah lumayan lengkap

safruls committed:
On Client side:
1. Write functions on main.js to enable social login using google.
2. Attach google login button on index.html
On server side:
1. Create a new route and static googleLogin method on userController.js
Other:
1. Update API documentation

safruls committed:
On client side:
1. Add button to add music track retrieved from Deezer (3rd party API) on main.js
2. Add function updateTrack on main to allow retrieved music tracks to be displayed on Todo's card
On server side:
1. Add migration file to insert a new column for music track.
2. Add route to allow access to Deezer (3rd party API)
3. Add 'track' attributes on Todo model.
Other:
1. Add a new route for accessing 3rd party API on API documentation.

safruls committed:
Revisi add todo di main.js dan Validasi due date di models/todo.js
